src/Sentry.Serilog/SentrySinkExtensions.cs

```diff
@@ -159,132 +205,45 @@ public static class SentrySinkExtensions
-        }
+        sentrySerilogOptions.Dsn = dsn ?? sentrySerilogOptions.Dsn;
+        sentrySerilogOptions.MinimumEventLevel = minimumEventLevel ?? sentrySerilogOptions.MinimumEventLevel;
+        sentrySerilogOptions.MinimumBreadcrumbLevel = minimumBreadcrumbLevel ?? sentrySerilogOptions.MinimumBreadcrumbLevel;
+        sentrySerilogOptions.FormatProvider = formatProvider ?? sentrySerilogOptions.FormatProvider;
+        sentrySerilogOptions.TextFormatter = textFormatter ?? sentrySerilogOptions.TextFormatter;
+        sentrySerilogOptions.SendDefaultPii = sendDefaultPii ?? sentrySerilogOptions.SendDefaultPii;
+        sentrySerilogOptions.IsEnvironmentUser = isEnvironmentUser ?? sentrySerilogOptions.IsEnvironmentUser;
+        sentrySerilogOptions.ServerName = NonNullOrWhiteSpaceOrDefault(serverName, sentrySerilogOptions.ServerName);
+        sentrySerilogOptions.AttachStacktrace = attachStackTrace ?? sentrySerilogOptions.AttachStacktrace;
+        sentrySerilogOptions.MaxBreadcrumbs = maxBreadcrumbs ?? sentrySerilogOptions.MaxBreadcrumbs;
+        sentrySerilogOptions.SampleRate = sampleRate ?? sentrySerilogOptions.SampleRate;
+        sentrySerilogOptions.Release = NonNullOrWhiteSpaceOrDefault(release, sentrySerilogOptions.Release);
+        sentrySerilogOptions.Environment = NonNullOrWhiteSpaceOrDefault(environment, sentrySerilogOptions.Environment);
+        sentrySerilogOptions.MaxQueueItems = maxQueueItems ?? sentrySerilogOptions.MaxQueueItems;
+        sentrySerilogOptions.ShutdownTimeout = shutdownTimeout ?? sentrySerilogOptions.ShutdownTimeout;
+        sentrySerilogOptions.DecompressionMethods = decompressionMethods ?? sentrySerilogOptions.DecompressionMethods;
+        sentrySerilogOptions.RequestBodyCompressionLevel = requestBodyCompressionLevel ?? sentrySerilogOptions.RequestBodyCompressionLevel;
+        sentrySerilogOptions.RequestBodyCompressionBuffered = requestBodyCompressionBuffered ?? sentrySerilogOptions.RequestBodyCompressionBuffered;
+        sentrySerilogOptions.Debug = debug ?? sentrySerilogOptions.Debug;
+        sentrySerilogOptions.DiagnosticLevel = diagnosticLevel ?? sentrySerilogOptions.DiagnosticLevel;
+        sentrySerilogOptions.ReportAssembliesMode = reportAssembliesMode ?? sentrySerilogOptions.ReportAssembliesMode;
+        sentrySerilogOptions.DeduplicateMode = deduplicateMode ?? sentrySerilogOptions.DeduplicateMode;
 
         // Serilog-specific items
-        if (initializeSdk.HasValue)
-        {
-            sentrySerilogOptions.InitializeSdk = initializeSdk.Value;
-        }
-
-        if (defaultTags?.Any() == true)
+        sentrySerilogOptions.InitializeSdk = dsn is not null;
+        if (defaultTags?.Count > 0)
         {
             foreach (var tag in defaultTags)
             {
                 sentrySerilogOptions.DefaultTags.Add(tag.Key, tag.Value);
             }
         }
+
+        string? NonNullOrWhiteSpaceOrDefault(string? value, string? defaultValue)
```

Comment:
Looks like it could be static?

```suggestion
        static string? NonNullOrWhiteSpaceOrDefault(string? value, string? defaultValue)
```